data/generate_data.py

```diff
@@ -0,0 +1,69 @@
+import scanpy as sc
+import numpy as np
+import scIB
+
+
+def get_adata_rand_batch(pca=False, n_top_genes=None, neighbors=False):
+    adata = sc.datasets.paul15()
+    adata.obs['celltype'] = adata.obs['paul15_clusters']
+    np.random.seed(42)
+    adata.obs['batch'] = np.random.randint(1, 5, adata.n_obs)
+    adata.obs['batch'] = adata.obs['batch'].astype(str)
+    adata.obs['batch'] = adata.obs['batch'].astype("category")
```

Comment:
can just string together `astype(str).astype("category")`, no?

Reply:
or just use category directly?

Reply:
If I don't convert to str, conversion into R via `rpy2` will throw an error. But I agree, a one-liner is nicer

Reply:
375d957